<launch>
	
	<arg name="sim" default="false" />

    <!-- Start PID nodes for controls -->
    <include file="$(find controls)/launch/pid.launch">
        <arg name="sim" value="$(arg sim)" />
    </include>

	<!-- Start controls node that republishes state to seperate topics -->
	<node pkg="controls" name="state_republisher" type="state_republisher.py" />

	<!-- Start controls node that listens to PID and outputs thruster allocations -->
	<node pkg="controls" name="thruster_controls" type="thruster_controls.py" output="screen">
		<param name="sim" value="$(arg sim)" />
	</node>

	<!-- Start desired_state node that processes inputs from motion planning
    and outputs setpoints to PID -->
	<node pkg="controls" name="desired_state" type="desired_state.py" output="screen" />
	
</launch>
